deploy/e2e-test-image/setuser.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python3
```

Comment:
Why do we make an extra Docker image for our e2e tests?

Ok, it is minimal and so more time and energy efficient.
On the other hand, it increases the complexity, requires to add an foreign, static, not auto updated script to our repository and prevents us from using a Docker image that is used in production and not CI/CD tested at this moment.
We could set `E2E_TEST_DOCKER_IMAGE` in the Makefile statically to [`openhpi/co_execenv_python:3.8`](https://github.com/openHPI/dockerfiles/tree/master/co_execenv_python/3.8).

What do you think?

Reply:
I was wondering the same, to be honest, and thought we switched to the "famous" `[openhpi/co_execenv_python:3.8](https://github.com/openHPI/dockerfiles/tree/master/co_execenv_python/3.8)` image some time ago. Therefore, I was surprised that we didn't.

For a moment, I also considered switching (especially after configuring everything took me some while) to have the same advantages you mentioned. Sure, having a minimal image is also nice, but I would assume the time and energy benefits are negligible.

My main reason to keep it was to have an independent development cycle for the Docker images and Poseidon. In the current situation, using the "released" version of our Dockerfiles would prevent the successful test of not being root or cause many e2e failures (the `/sbin/setuser` script can only be executed by root). 
An alternative to adding the Python script included in our Images by default would be to change the `FROM` directive in the Dockerfile: Instead of starting with a [Python-Image](https://github.com/openHPI/poseidon/blob/01e5ed0d4e2b6fec9c7738426d4f9c4e9b4305fd/deploy/e2e-test-image/Dockerfile#L2) and adding the `/sbin/setuser` script, we could bring it closer to our production environment:

```dockerfile
FROM phusion/baseimage:master
```

This would allow us to remove the "arbitrary" Python script from the folder, while still keeping an independent Docker file for testing purposes in this repository. However, by bringing that closer, we also reduce the argument not to use the production environment...

What would you prefer?

Reply:
I see the argument of having an independent development cycle. But this means, that we have to do the changes to the Dockerfiles twice - once in the other repo and once here. If we have to do this anyways we could also:

1. Have `FROM openhpi/co_execenv_python:3.8` as default.
2. Going down the base-image-chain as far as required for development and copy the changes to this repo.
3. Merge the [Dockerfiles Repo](https://github.com/openHPI/dockerfiles) first.
4. Return to our default (`FROM openhpi/co_execenv_python:3.8`) here and merge this.

I assume this will not be required often. Maybe it's too complicated? I'm also fine with your solution.

Reply:
I updated the GitHub action workflow to checkout a *matching* branch name if possible. 
- With a non-matching branch, the published Docker image is used (see example run with branch "test" [here](https://github.com/openHPI/poseidon/actions/runs/3100012999))
- With a matching brach, the not-yet-merged Dockerfile is used to build a new image (see example run with branch "start_init_scripts [here](https://github.com/openHPI/poseidon/actions/runs/3100015284))

Two additional things to note:
- I switched the Python:3.8 image in favour of Java:17, as our tests require the image to contain the `make` command which is not available in the Python:3.8 image
- The remote branch in the docker files repository has a different name. Unfortunately, I cannot rename that branch remotely without closing my existing MR there. Hence, I duplicated the branch to match the name of this branch and will take care of deleting the duplicated branch once we merged everything.